src/librustc_mir/interpret/place.rs

```diff
@@ -684,9 +684,16 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M> {
         layout: TyLayout<'tcx>,
         kind: MemoryKind<M::MemoryKinds>,
     ) -> EvalResult<'tcx, MPlaceTy<'tcx>> {
-        assert!(!layout.is_unsized(), "cannot alloc memory for unsized type");
-        let ptr = self.memory.allocate(layout.size, layout.align, kind)?;
-        Ok(MPlaceTy::from_aligned_ptr(ptr, layout))
+        if layout.is_unsized() {
+            assert!(self.tcx.features().unsized_locals, "cannot alloc memory for unsized type");
+            // allocate a fat pointer slot instead
```

Comment:
Wait what? That seems wrong. We can have proper unsized locals, why should we have these indirections? Does that match how the MIR looks?

Reply:
I assumed we'd want to mirror rustc codegen

Reply:
I don't think so. We already have the ability to use an `Operand::Indirect` with a `MemPlace` to store the metadata -- that's like a properly typed version of a fat pointer.